components/brave_wallet/browser/keyring_controller.cc

```diff
@@ -770,12 +774,9 @@ void KeyringController::AddHardwareAccounts(
   const auto hash = base::PersistentHash(infos.front()->address);
   std::string device_id = hardware_vendor + std::to_string(hash);
 
-  base::Value hardware_keyrings(base::Value::Type::DICTIONARY);
-  const base::Value* value = GetPrefForHardwareKeyring(prefs_);
-  if (value) {
-    hardware_keyrings = value->Clone();
-  }
-  const base::Value* device_value = hardware_keyrings.FindKey(device_id);
+  base::Value* hardware_keyrings = GetPrefForHardwareKeyringUpdate(prefs_);
+
+  const base::Value* device_value = hardware_keyrings->FindKey(device_id);
   base::Value device_keyring(base::Value::Type::DICTIONARY);
   if (device_value)
     device_keyring = device_value->Clone();
```

Comment:
we don't need Clone here since the top level base::Value is mutable now, same as `meta_value->Clone()`

Reply:
done